generate_cache.py

```diff
@@ -212,10 +217,12 @@ def get_specs_metadata(specs: dict[str, list[spack.spec.Spec]]) -> dict:
                     if key == "target" and isinstance(value, dict):
                         value = "%s %s" % (value["vendor"], value["name"])
 
-                compiler = "%s@%s" % (
-                    spec["compiler"]["name"],
-                    spec["compiler"]["version"],
-                )
+                try:
```

Comment:
Maybe this could be a function just in case the compiler name/format changes for some reason?